FreeRTOS_Sockets.c

```diff
@@ -2958,14 +3267,13 @@ uint32_t FreeRTOS_inet_addr( const char * pcIPAddress )
         uint8_t * pucTarget = pvDestination;
         const char * pcIterator = pcSource;
 
-
         ( void ) memset( pucTarget, 0, ipSIZE_OF_IPv6_ADDRESS );
         xTargetIndex = 0;
         xColon = -1;
         xResult = 0;
 
         /* Leading :: requires some special handling. */
-        if( ( pcIterator[ 0 ] == ':' ) && ( pcIterator[ 1 ] == ':' ) && ( pcIterator[ 2 ] == '\0' ) )
+        if( memcmp( pcIterator, "::", 3 ) == 0 )
```

Comment:
MISRA Rule 21.14 requires that memcmp should not be used on null terminated strings.

Reply:
This PR was already created 19 days ago and in a meanwhile I worked on both MISRA and the complexity of other functions.
~~~c
-    if( memcmp( pcIterator, "::", 3 ) == 0 )
+    if( strcmp( pcIterator, "::" ) == 0 )
~~~
Let us first finish and merge this PR236, and then I will create a new PR to get all complexity scores below 9.

Reply:
Here is a question about the "misra-c-2012-rule-15-4" rule on [stackoverflow](https://stackoverflow.com/questions/66369536/misra-c-2012-rule-15-4-and-replacing-gotos-with-breaks).
It says that "But it is just an **advisory** one and there's nothing wrong with using multiple breaks like you do as long as the code is readable and easy to follow".